public/components/datasources/components/manage/associated_objects_tab.tsx

```diff
@@ -39,7 +47,23 @@ export const AssociatedObjectsTab: React.FC<AssociatedObjectsTabProps> = ({
 
   useEffect(() => {
     fetchAssociatedObjects();
-  }, []);
+
+    const databaseOptions = [...new Set(associatedObjects.map((obj) => obj.database))]
```

Comment:
Are you using `...new Set()` to remove duplicates from `associatedObjects.map((obj) => obj.database)`? If so, maybe consider using lodash's `_.uniq()` method so that we can use something meant for removing duplicates and remove ambiguity. There are also some other removing duplicates lodash methods that can be used for performance improvement using attributes of the values.

Reply:
Similarly in line 57